medusa/clients/torrent/utorrent_client.py

```diff
@@ -54,17 +56,63 @@ def _get_auth(self):
             return self.auth
 
     def _add_torrent_uri(self, result):
-        return self._request(params={
+            ## Set proper subfoler as destination for uTorrent 
+            series_id = result.series.series_id
+            series_name = result.series.name
+ 
+			# Get the sub-folder the user has assigned to that series
+            root_dirs = app.ROOT_DIRS
+            root_location = root_dirs[int(root_dirs[0]) + 1]
+            torrent_path = result.series._location
+            torrent_subfolder = None
+
+            if not root_location == torrent_path:
+                torrent_subfolder = os.path.basename(torrent_path)
+	        ## Use the label if tyhere is no subfolder
+            else:
+                torrent_subfolder = series_name
+
+            log.info('Show {name}: torrent snatched, download destination folder is: {path} (sub-folder: {sub})',
+                    {'name': series_name, 'path': torrent_path, 'sub': torrent_subfolder})
+
+
+            return self._request(params={
             'action': 'add-url',
             # limit the param length to 1024 chars (uTorrent bug)
             's': result.url[:1024],
+			## add torrent path to request
+			#   
+			'path': torrent_subfolder,
         })
 
     def _add_torrent_file(self, result):
-        return self._request(
-            method='post',
-            params={
-                'action': 'add-file',
+            ##Set the series subfoler as  download destination for uTorrent 
+            series_id = result.series.series_id
+            series_name = result.series.name
+ 
+			##Get the sub-folder the user has assigned to that series
+            root_dirs = app.ROOT_DIRS
+            root_location = root_dirs[int(root_dirs[0]) + 1]
+            torrent_path = result.series._location
+
+            torrent_subfolder = None;
+            if not root_location == torrent_path:
+                torrent_subfolder = os.path.basename(torrent_path)
+	        ## Use the label if tyhere is no subfolder
+            else:
+                torrent_subfolder = series_name
+
+            log.info('Show {name}: torrent snatched, download destination folder is: {path} (sub-folder: {sub})',
+                    {'name': series_name, 'path': torrent_path, 'sub': torrent_subfolder})
```

Comment:
This code is exactly the same as above and should be put in a single method called for example: `get_torrent_subfolder()`.

Reply:
Done.